crud.py

```diff
@@ -0,0 +1,57 @@
+from sqlalchemy.orm import Session
+
+import models, schemas
+
+
+def get_all_books(
+    db: Session,
+    author_id: int | None = None,
+    skip: int = 0, limit: int = 10
+):
+    queryset = db.query(models.DBBook)
+    if author_id:
+        queryset = queryset.filter(models.DBBook.author_id == author_id)
+    return queryset.offset(skip).limit(limit).all()
+
+
+def get_book(db: Session, book_id: int):
+    return db.query(models.DBBook).filter(models.DBBook.id == book_id).first()
+
+def get_book_by_author_id(db: Session, author_id: int):
+    return db.query(models.DBBook).filter(models.DBBook.author_id == author_id).all()
+
+
+def create_book(db: Session, book: schemas.BookCreate):
+    db_book = models.DBBook(
+        title=book.title,
+        summary=book.summary,
+        publication_date=book.publication_date,
+        author_id=book.author_id,
+    )
+
+    db.add(db_book)
+    db.commit()
+    db.refresh(db_book)
+
+    return book

```

Comment:
The `create_book` function returns the `book` parameter, which is an instance of `schemas.BookCreate`. It might be more useful to return the `db_book` object, which is the actual database model instance that was added and committed to the session. This would provide the caller with the complete database object, including any auto-generated fields like `id`.